examples/dbscan/ArborX_DBSCAN.hpp

```diff
@@ -236,60 +236,70 @@ void dbscan(ExecutionSpace exec_space, Primitives const &primitives,
   start = clock::now();
   Kokkos::Profiling::pushRegion("ArborX::DBSCAN::sort_and_filter_clusters");
 
+  Kokkos::View<int *, MemorySpace> cluster_sizes(
+      "ArborX::DBSCAN::cluster_sizes", n);
+  Kokkos::parallel_for("ArborX::DBSCAN::compute_cluster_sizes",
+                       Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, n),
+                       KOKKOS_LAMBDA(int const i) {
+                         Kokkos::atomic_fetch_add(&cluster_sizes(clusters(i)),
+                                                  1);
+                       });
+
+  int num_skipped = 0;
+  Kokkos::parallel_reduce("ArborX::DBSCAN::replace_skipped_cluster_indices",
+                          Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, n),
+                          KOKKOS_LAMBDA(int const i, int &update) {
+                            if (cluster_sizes(clusters(i)) < cluster_min_size)
+                            {
+                              clusters(i) = INT_MAX;
+                              update++;
+                            }
+                          },
+                          num_skipped);
+  auto num_cluster_indices = n - num_skipped;
+
   // sort clusters and compute permutation
   auto permute = Details::sortObjects(exec_space, clusters);
```

Comment:
In this sort, all indices that correspond to small clusters, will be at the end of `clusters` as we set them to `INT_MAX` above. Thus, we can simply truncate `permute` to give us `cluster_indices`.

Reply:
Comment why you INT_MAX